qa/workunits/rbd/rbd_groups.sh

```diff
@@ -210,15 +210,31 @@ check_snapshot_info()
     local snap_name=$2
     local image_count=$3
 
-    local snap_info=$(rbd group snap info $group_name@$snap_name --format=json)
-    local actual_snap_name=$(jq -r ".name" <<< "$snap_info")
+    local snap_info_json=$(
+        rbd group snap info $group_name@$snap_name --format=json)
+    local actual_snap_name=$(jq -r ".name" <<< "$snap_info_json")
     test "$actual_snap_name" = "$snap_name" || return 1
 
-    local snap_state=$(jq -r ".state" <<< "$snap_info")
+    local snap_state=$(jq -r ".state" <<< "$snap_info_json")
     test "$snap_state" = "complete" || return 1
 
-    local actual_image_count=$(jq '.images | length' <<< "$snap_info")
-    test "$actual_image_count" = "$image_count"
+    local actual_image_count=$(jq '.images | length' <<< "$snap_info_json")
+    test "$actual_image_count" = "$image_count" || return 1
+
+    local image_snap_name_json=$(
+        jq -r '.image_snap_name' <<< "$snap_info_json")
+    local snap_info=$(rbd group snap info $group_name@$snap_name)
```

Comment:
This swallows `rbd group snap info` errors -- if it fails (i.e. doesn't work at all) and `image_count` is 0, that would go unnoticed because we only test for emptiness there.  I'd suggest adding a grep for `state:` and asserting that it comes back with `state: complete` below to make sure that some valid output is always produced.